examples/virtual_methods/derived_button/imp.rs

```diff
@@ -0,0 +1,39 @@
+use crate::base_button::*;
+use gtk::{gio, glib, prelude::*, subclass::prelude::*};
+use std::{future::Future, pin::Pin};
+
+#[derive(Debug)]
+pub struct DerivedButton {}
+
+#[glib::object_subclass]
+impl ObjectSubclass for DerivedButton {
+    const NAME: &'static str = "ExampleDerivedButton";
+    type ParentType = BaseButton;
+    type Type = super::DerivedButton;
+
+    fn new() -> Self {
+        Self {}
+    }
```

Comment:
```suggestion
```
same as other private structs, the default new function calls Default::default()